src/ophyd_epics_devices/zebra.py

```diff
@@ -0,0 +1,306 @@
+from enum import Enum
+from functools import partialmethod
+from typing import List, Type
+
+import numpy as np
+import numpy.typing as npt
+from ophyd.v2.core import StandardReadable, T
+from ophyd.v2.epics import EpicsSignalR, EpicsSignalRW
+
+
+def zebra_rw(datatype: Type[T], suffix: str) -> EpicsSignalRW[T]:
+    return EpicsSignalRW(datatype, suffix)
+
+
+def zebra_r(datatype: Type[T], suffix: str) -> EpicsSignalR[T]:
+    return EpicsSignalR(datatype, suffix)
+
+
+class Active(Enum):
+    no = "No"
+    yes = "Yes"
+
+
+class GatePulseSelect(Enum):
+    time = "Time"
+    position = "Position"
+    external = "External"
+
+
+class ArmSelect(Enum):
+    soft = "Soft"
+    external = "External"
+
+
+class Direction(Enum):
+    positive = "Positive"
+    negative = "Negative"
+
+
+class EncoderType(Enum):
+    enc1 = "Enc1"
+    enc2 = "Enc2"
+    enc3 = "Enc3"
+    enc4 = "Enc4"
+    enc1_4av = "Enc1-4Av"
+
+
+class TimeUnits(Enum):
+    ms = "ms"
+    s = "s"
+    s10 = "10s"
+
+
+class Capture(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        self.enc1 = zebra_rw(Active, "B0")
+        self.enc2 = zebra_rw(Active, "B1")
+        self.enc3 = zebra_rw(Active, "B2")
+        self.enc4 = zebra_rw(Active, "B3")
+        self.sys1 = zebra_rw(Active, "B4")
+        self.sys2 = zebra_rw(Active, "B5")
+        self.div1 = zebra_rw(Active, "B6")
+        self.div2 = zebra_rw(Active, "B7")
+        self.div3 = zebra_rw(Active, "B8")
+        self.div4 = zebra_rw(Active, "B9")
+        super().__init__(prefix, name)
+
+
+class ArrayOuts(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        def make_pv(suffix: str, common_prefix: str = "PC_"):
+            return zebra_rw(npt.NDArray[np.float64], common_prefix + suffix)
+
+        self.enc1 = make_pv("ENC1")
+        self.enc2 = make_pv("ENC2")
+        self.enc3 = make_pv("ENC3")
+        self.enc4 = make_pv("ENC4")
+        self.sys1 = make_pv("SYS1")
+        self.sys2 = make_pv("SYS2")
+        self.div1 = make_pv("DIV1")
+        self.div2 = make_pv("DIV2")
+        self.div3 = make_pv("DIV3")
+        self.div4 = make_pv("DIV4")
+        super().__init__(prefix, name)
+
+
+class ZebraOutputPanel(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        self.pulse_1_input = zebra_rw(int, "PULSE1_INP")
+
+        self.out_1 = zebra_rw(int, "OUT1_TTL")
+        self.out_2 = zebra_rw(int, "OUT2_TTL")
+        self.out_3 = zebra_rw(int, "OUT3_TTL")
+        self.out_4 = zebra_rw(int, "OUT4_TTL")
```

Comment:
Should: Now that we're no longer using `Component` in the same way I think we can jut make `out_pvs` in the constructor and not have to have out_1, 2 etc.

Reply:
Sorry, I'm not sure what you mean. Do you mean, we should make an array like this:

```
class ZebraOutputPanel(StandardReadable):
    def __init__(self, prefix, name):
        ...
        self.out_pvs = [zebra_rw(int, "OUT1_TTL"), zebra_rw(int, "OUT2_TTL"), zebra_rw(int, "OUT3_TTL"), zebra_rw(int, "OUT4_TTL")]
```

or something similar? i.e. what is the issue with having self.out_1/ out_2 explicitly defined and used?

Reply:
Well something like this is tidier:
```python
class ZebraOutputPanel(StandardReadable):
    def __init__(self, prefix, name):
        ...
        self.out_pvs = [zebra_rw(int, f"OUT{channel}_TTL") for channel in range (1, 4)]
```
But there's no issue, it's fine. I just don't like having two ways of accessing things. It seems to me that a list of them is the natural way to do it, the only reason I ever had `out_1` is 'cos making a list wasn't possible before. We never use them explicitly in our code.

Reply:
For PandA I did something similar, but using dicts, so here it would be:
```python
class ZebraOutputPanel(StandardReadable):
    def __init__(self, prefix, name):
        ...
        self.out_ttl = {channel: zebra_rw(int, f"OUT{channel}_TTL") for channel in range (1, 4)}
```
Then you can reference `zebra.out_ttl[1]` to get `OUT1_TTL`

Reply:
I also think `ttl` is better than `pv`, it's redundant to label everything as pvs